metron-interface/metron-alerts/src/app/alerts/alerts-list/query-builder.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Filter} from '../../model/filter';
+import {ColumnNamesService} from '../../service/column-names.service';
+import {SearchRequest} from '../../model/search-request';
+
+export class QueryBuilder {
+  private _searchRequest = new SearchRequest();
+  private _query = '*';
+  private _displayQuery = this._query;
+  private _filters: Filter[] = [];
+
+  set query(value: string) {
+    value = value.replace(/\\:/g, ':');
+    this._query = value;
+    this.updateFilters(this._query, false);
+    this.onSearchChange();
+  }
+
+  get query(): string {
+    return this._query;
+  }
+
+  set displayQuery(value: string) {
+    this._displayQuery = value;
+    this.updateFilters(this._displayQuery, true);
+    this.onSearchChange();
+  }
+
+  get displayQuery(): string {
+    return this._displayQuery;
+  }
+
+  get filters(): Filter[] {
+    return this._filters;
+  }
+
+
+  get searchRequest(): SearchRequest {
+    this._searchRequest.query = { query_string: { query: this.generateSelect() } };
```

Comment:
This looks specific to Elasticsearch.  I would expect:
`this._searchRequest.query = this.generateSelect()`